components/script/script_module.rs

```diff
@@ -652,24 +652,109 @@ impl ModuleTree {
         Ok(specifier_urls)
     }
 
-    /// The following module specifiers are allowed by the spec:
-    ///  - a valid absolute URL
-    ///  - a valid relative URL that starts with "/", "./" or "../"
-    ///
-    /// Bareword module specifiers are currently disallowed as these may be given
-    /// special meanings in the future.
     /// <https://html.spec.whatwg.org/multipage/#resolve-a-module-specifier>
     #[allow(unsafe_code)]
     fn resolve_module_specifier(
-        cx: *mut JSContext,
-        url: &ServoUrl,
-        specifier: RawHandle<*mut JSString>,
-    ) -> Option<ServoUrl> {
-        let specifier_str = unsafe { jsstring_to_str(cx, ptr::NonNull::new(*specifier).unwrap()) };
+        global: &GlobalScope,
+        script: Option<&ModuleScript>,
+        specifier: DOMString,
+        can_gc: CanGc,
+    ) -> Fallible<ServoUrl> {
+        // Step 1~3 to get settingsObject and baseURL
+        let script_global = script.and_then(|s| s.owner.as_ref().map(|o| o.global()));
+        let (global, base_url): (&GlobalScope, &ServoUrl) = match script {
+            Some(s) => (script_global.as_ref().map_or(global, |g| g), &s.base_url),
+            None => (global, &global.api_base_url()),
```

Comment:
Note that I'm not sure if `global` here is the [`current-settings-object`](https://html.spec.whatwg.org/multipage/webappapis.html#current-settings-object). 

For example, [here](https://github.com/servo/servo/blob/781c2f4a5671c2b7eb3e794c735e42cdda74ac5b/components/script/script_module.rs#L1248) it seems to be the global of the owner, which is not necessarily the same thing. 

Note sure how to ensure we have the right global. You may need to use `enter_realm` at the original call point. 

Would be good to add a note and file an issue to check this. 

cc @jdm

Reply:
I'll list it in #37553